### PR TITLE
remove {'source': ''} from every error

### LINKED ISSUE #6971: Remove {'source': ''} from every error
It is extraordinarily redundant and does not even serve its purpose. Change it to actual source of error or remove it

### PR DESCRIPTION
**Pull request by satya7289**
Fixes #6971

#### Short description of what this resolves:
Change the actual source of error and remove source if not necessary.

#### Changes proposed in this pull request:

- update or remove the {'source':''}

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           

Clones added
============
- app/api/role_invites.py  2
         

Clones removed
==============
+ app/api/attendees.py  -1
+ app/api/orders.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5562667&bid=18242282)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           

Clones added
============
- app/api/role_invites.py  2
         

Clones removed
==============
+ app/api/attendees.py  -1
+ app/api/orders.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5562667&bid=18242282)

**Comment by iamareebjamal** [comment]
First complete #7019

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           

Clones added
============
- app/api/role_invites.py  2
         

Clones removed
==============
+ app/api/orders.py  -2
+ app/api/attendees.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5562667&bid=18242282)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           

Clones added
============
- app/api/role_invites.py  2
         

Clones removed
==============
+ app/api/attendees.py  -1
+ app/api/orders.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5562667&bid=18242282)

**Comment by satya7289** [comment]
@iamareebjamal pls let me know what changes need to make...

File: app/api/access_codes.py [hunk 1]
```diff
@@ -36,7 +36,8 @@ def before_post(self, args, kwargs, data):
         """
         require_relationship(['event', 'user'], data)
         if not has_access('is_coorganizer', event_id=data['event']):
-            raise ForbiddenError({'source': ''}, "Minimum Organizer access required")
+            raise ForbiddenError({'parameter': 'event_id'},
```
Comment (iamareebjamal) [inline app/api/access_codes.py:39]: There is no parameter `event_id`, change this to `pointer: /data/event`

File: app/api/access_codes.py [hunk 2]
```diff
@@ -93,12 +94,12 @@ def query(self, view_kwargs):
...
             user = safe_query_kwargs(User, view_kwargs, 'user_id')
             if not has_access('is_user_itself', user_id=user.id):
-                raise ForbiddenError({'source': ''}, 'Access Forbidden')
+                raise ForbiddenError({'parameter': 'user_id'}, 'Access Forbidden')
             query_ = query_.join(User).filter(User.id == user.id)
         if view_kwargs.get('ticket_id'):
             ticket = safe_query_kwargs(Ticket, view_kwargs, 'ticket_id')
             if not has_access('is_coorganizer', event_id=ticket.event_id):
-                raise ForbiddenError({'source': ''}, 'Access Forbidden')
+                raise ForbiddenError({'parameter': 'event_id'}, 'Access Forbidden')
```
Comment (satya7289) [inline app/api/access_codes.py:102]: In this situation what will be source or pointer? ticket_id or event_id or user_id..!!

File: app/api/access_codes.py (hunk 1 above)
Comment (satya7289) [reply app/api/access_codes.py:39]: ok

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           

Clones added
============
- app/api/role_invites.py  2
         

Clones removed
==============
+ app/api/attendees.py  -1
+ app/api/orders.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5562667&bid=18242282)

File: app/api/access_codes.py (hunk 2 above)
Comment (iamareebjamal) [reply app/api/access_codes.py:102]: ticket_id, paramter

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           

Clones added
============
- app/api/role_invites.py  2
         

Clones removed
==============
+ app/api/orders.py  -2
+ app/api/attendees.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5562667&bid=18242282)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           

Clones added
============
- app/api/role_invites.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5562667&bid=18242282)

**Comment by satya7289** [comment]
@iamareebjamal pls let me know what other things need to be modified.

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           

Clones added
============
- app/api/role_invites.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5562667&bid=18242282)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           

Clones added
============
- app/api/role_invites.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5562667&bid=18242282)

File: app/api/access_codes.py [hunk 3]
```diff
@@ -160,9 +161,8 @@ def before_get(self, args, kwargs):
                 raise ObjectNotFound({'parameter': '{id}'}, "Access Code:  not found")
 
             if not has_access('is_coorganizer', event_id=access.event_id):
-                raise UnprocessableEntityError(
-                    {'source': ''}, "Please verify your permission"
-                )
+                raise UnprocessableEntityError({'parameter': 'id'},
```
Comment (iamareebjamal) [inline app/api/access_codes.py:164]: Remove paramater id, just second argument is enough

File: app/api/access_codes.py [hunk 4]
```diff
@@ -93,12 +94,12 @@ def query(self, view_kwargs):
...
             user = safe_query_kwargs(User, view_kwargs, 'user_id')
             if not has_access('is_user_itself', user_id=user.id):
-                raise ForbiddenError({'source': ''}, 'Access Forbidden')
+                raise ForbiddenError({'parameter': 'user_id'}, 'Access Forbidden')
             query_ = query_.join(User).filter(User.id == user.id)
         if view_kwargs.get('ticket_id'):
             ticket = safe_query_kwargs(Ticket, view_kwargs, 'ticket_id')
             if not has_access('is_coorganizer', event_id=ticket.event_id):
-                raise ForbiddenError({'source': ''}, 'Access Forbidden')
+                raise ForbiddenError({'parameter': 'ticket_id'}, 'Access Forbidden')
```
Comment (iamareebjamal) [inline app/api/access_codes.py:102]: Make it `ticket.event`

File: app/api/access_codes.py [hunk 5]
```diff
@@ -93,12 +94,12 @@ def query(self, view_kwargs):
         if view_kwargs.get('user_id'):
             user = safe_query_kwargs(User, view_kwargs, 'user_id')
             if not has_access('is_user_itself', user_id=user.id):
-                raise ForbiddenError({'source': ''}, 'Access Forbidden')
+                raise ForbiddenError({'parameter': 'user_id'}, 'Access Forbidden')
```
Comment (iamareebjamal) [inline app/api/access_codes.py:97]: access_code

File: app/api/attendees.py [hunk 6]
```diff
@@ -135,7 +135,8 @@ def query(self, view_kwargs):
             if not has_access('is_registrar', event_id=order.event_id) and not has_access(
                 'is_user_itself', user_id=order.user_id
             ):
-                raise ForbiddenError({'source': ''}, 'Access Forbidden')
+                raise ForbiddenError({'parameter': 'order_identifier'},
```
Comment (iamareebjamal) [inline app/api/attendees.py:138]: order.event

File: app/api/attendees.py [hunk 7]
```diff
@@ -147,7 +148,7 @@ def query(self, view_kwargs):
         if view_kwargs.get('user_id'):
             user = safe_query_kwargs(User, view_kwargs, 'user_id')
             if not has_access('is_user_itself', user_id=user.id):
-                raise ForbiddenError({'source': ''}, 'Access Forbidden')
+                raise ForbiddenError({'parameter': 'user_id'}, 'Access Forbidden')
```
Comment (iamareebjamal) [inline app/api/attendees.py:151]: attendee

File: app/api/auth.py [hunk 8]
```diff
@@ -196,7 +196,8 @@ def login_user(provider):
             'code': request.args.get('code'),
         }
         if not payload['client_id'] or not payload['client_secret']:
-            raise NotImplementedError({'source': ''}, 'Facebook Login Not Configured')
+            raise NotImplementedError({'source': 'client_id or client_secret'},
```
Comment (iamareebjamal) [inline app/api/auth.py:199]: remove source

File: app/api/auth.py [hunk 9]
```diff
@@ -278,19 +279,19 @@ def login_user(provider):
 def verify_email():
     try:
         token = base64.b64decode(request.json['data']['token'])
-    except base64.binascii.Error:
-        raise BadRequestError({'source': ''}, 'Invalid Token')
+    except base64.binascii.Error as e:
+        raise BadRequestError({'source': e}, 'Invalid Token')
```
Comment (iamareebjamal) [inline app/api/auth.py:283]: Source is token

File: app/api/auth.py [hunk 10]
```diff
@@ -278,19 +279,19 @@ def login_user(provider):
...
     s = get_serializer()
 
     try:
         data = s.loads(token)
-    except Exception:
-        raise BadRequestError({'source': ''}, 'Invalid Token')
+    except Exception as e:
+        raise BadRequestError({'source': e}, 'Invalid Token')
```
Comment (iamareebjamal) [inline app/api/auth.py:289]: token

File: app/api/auth.py [hunk 11]
```diff
@@ -278,19 +279,19 @@ def login_user(provider):
...
 
     try:
         user = User.query.filter_by(email=data[0]).one()
-    except Exception:
-        raise BadRequestError({'source': ''}, 'Invalid Token')
+    except Exception as e:
+        raise BadRequestError({'source': e}, 'Invalid Token')
```
Comment (iamareebjamal) [inline app/api/auth.py:294]: token

File: app/api/auth.py [hunk 12]
```diff
@@ -301,14 +302,14 @@ def verify_email():
 def resend_verification_email():
     try:
         email = request.json['data']['email']
-    except TypeError:
-        raise BadRequestError({'source': ''}, 'Bad Request Error')
+    except TypeError as e:
+        raise BadRequestError({'source': e}, 'Bad Request Error')
```
Comment (iamareebjamal) [inline app/api/auth.py:306]: Source is email

**Comment by iamareebjamal** [comment]
Make similar changes as pointed out

File: app/api/access_codes.py (hunk 4 above)
Comment (satya7289) [reply app/api/access_codes.py:102]: here `'source': 'ticket.event'` will be right !!

File: app/api/access_codes.py (hunk 3 above)
Comment (satya7289) [reply app/api/access_codes.py:164]: Here only `'source': 'id'` will be fine!!

File: app/api/auth.py (hunk 11 above)
Comment (satya7289) [reply app/api/auth.py:294]: Should I remove all `'source': e` to actual error message like token and all!!

Comment (iamareebjamal) [reply app/api/auth.py:294]: Just invalid token would be fine

File: app/api/speakers_calls.py [hunk 13]
```diff
@@ -94,11 +95,11 @@ def before_patch(self, args, kwargs, data):
                     or speakers_call.ends_at > event.starts_at
                 ):
                     raise ForbiddenError(
-                        {'source': ''},
+                        {'parameter': 'event_id'},
                         "Speakers call date can\'t be after the event start date",
                     )
-            except NoResultFound:
-                raise ObjectNotFound({'source': ''}, "Object: not found")
+            except NoResultFound as e:
```
Comment (niranjan94) [inline app/api/speakers_calls.py:101]: Codacy found an issue: [Unused variable 'e'](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5562667)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Clones removed
==============
+ app/api/orders.py  -2
+ app/api/access_codes.py  -1
+ app/api/attendees.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5562667&bid=18242282)

**Comment by maze-runnar** [comment]
just have to resolve conflict, and this PR is good to merge.

**Pull request closed without merging**